test/beacon/Voucher.test.ts

```diff
@@ -48,26 +77,51 @@ describe('Voucher', function () {
             requester,
             anyone,
         ] = await ethers.getSigners();
-        const beacon = await voucherUtils.deployBeaconAndImplementation(admin.address);
+        // Deploy PoCo mock and VoucherHub.
         iexecPocoInstance = await new IexecPocoMock__factory()
             .connect(admin)
             .deploy()
             .then((x) => x.waitForDeployment());
         iexecPoco = await iexecPocoInstance.getAddress();
-        const voucherHub = await voucherHubUtils.deployHub(
+        beacon = await voucherUtils.deployBeaconAndImplementation(admin.address);
+        voucherHub = await voucherHubUtils.deployHub(
             assetEligibilityManager.address,
             voucherManager.address,
             iexecPoco,
             await beacon.getAddress(),
         );
-        voucherHubWithVoucherManagerSigner = voucherHub.connect(voucherManager);
-        voucherHubWithAssetEligibilityManagerSigner = voucherHub.connect(assetEligibilityManager);
-        voucherHubWithAnyoneSigner = voucherHub.connect(anyone);
-        await voucherHubWithAssetEligibilityManagerSigner.createVoucherType(description, duration);
-
+        // Fund VoucherHub with RLCs.
         await iexecPocoInstance
             .transfer(await voucherHub.getAddress(), initVoucherHubBalance)
             .then((tx) => tx.wait());
+        // TODO rename to voucherHubAs...
+        voucherHubAsVoucherCreationManager = voucherHub.connect(voucherManager);
+        voucherHubAsAssetEligibilityManager = voucherHub.connect(assetEligibilityManager);
```

Comment:
Same here => move on top of line 94 ?

Reply:
I agree, it makes more sense to init all VoucherHub instances at once.
[Create different VoucherHub instances at the same place](https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/18/commits/b0ed7e02aabf7b5880a9153ba71388b0883d2ed3)